File: backend/models/herb.py
from sqlalchemy import Column, Integer, String, Float, Text
from core.database import Base


class Herb(Base):
    __tablename__ = "herb"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(64), unique=True, nullable=False, index=True)
    category = Column(String(50), nullable=False)
    nature = Column(String(20))
    taste = Column(String(20))
    meridian = Column(String(100))
    min_dosage = Column(Float)
    max_dosage = Column(Float)
    unit = Column(String(5), default="g")
    is_toxic = Column(Integer, default=0)
    functions = Column(Text)
    indications = Column(Text)

File: frontend/src/types/herb.ts
export interface Herb {
  id: number
  name: string
  category: string
  nature: string
  taste: string
  meridian: string
  minDosage: number
  maxDosage: number
  unit: string
  toxic: boolean
  functions: string[]
}

export interface SelectedHerb extends Herb {
  currentDosage: number
}
